src/test/datascience/jupyterServer.node.ts

```diff
@@ -288,11 +284,37 @@ export class JupyterServer {
                         }
                     }
                 };
+                let allOutput = '';
                 const subscription = result.out.subscribe((output) => {
+                    allOutput += output.out;
+
                     // When debugging Web Tests using VSCode dfebugger, we'd like to see this info.
                     // This way we can click the link in the output panel easily.
-                    if (output.out.indexOf('Use Control-C to stop this server and shut down all kernels')) {
-                        resolve(procDisposable);
+                    if (output.out.indexOf('Use Control-C to stop this server and shut down all kernels') >= 0) {
+                        const lines = splitLines(allOutput, { trim: true, removeEmptyEntries: true });
+                        const indexOfCtrlC = lines.findIndex((item) =>
+                            item.includes('Use Control-C to stop this server')
+                        );
+                        const lineWithUrl = lines
+                            .slice(0, indexOfCtrlC)
+                            .reverse()
+                            .find(
+                                (line) =>
+                                    line.includes(`http://localhost:${port}`) ||
+                                    line.includes(`https://localhost:${port}`)
+                            );
+                        let url = '';
+                        if (lineWithUrl) {
+                            url = lineWithUrl.substring(lineWithUrl.indexOf('http'));
+                        } else {
+                            url = `http${useCert ? 's' : ''}://localhost:${port}/?token=${token}`;
+                        }
+                        // token might not be printed in the output
+                        if (url.includes(`token=...`)) {
+                            url = url.replace(`token=...`, `token=${token}`);
+                        }
+                        console.log(`Started Jupyter Server on ${url}`);
```

Comment:
Did a `console.log` slip through?

Reply:
Was deliberate, (moved from other places to this one place) 
This code path is hit only on ci when launching tests